internal/controller/nonadminbackup_controller.go

```diff
@@ -97,11 +144,279 @@ func (r *NonAdminBackupReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+		if err := r.Delete(ctx, nab); err != nil {
+			logger.Error(err, "Failed to call Delete on the NonAdminBackup object")
+			return false, err
+		}
+		return true, nil // Requeue to allow deletion to proceed
+	}
+	return false, nil
+}
+
+// setStatusForDirectKubernetesApiDeletion updates the status and conditions when a NonAdminBackup
+// is deleted directly through the Kubernetes API. Only updates status and conditions
+// if the NAB finalizer exists.
+//
+// Parameters:
+//   - ctx: Context for managing request lifetime
+//   - logger: Logger instance
+//   - nab: NonAdminBackup being deleted
+//
+// Returns:
+//   - bool: true if status was updated
+//   - error: any error encountered
+func (r *NonAdminBackupReconciler) setStatusForDirectKubernetesApiDeletion(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	if controllerutil.ContainsFinalizer(nab, constant.NabFinalizerName) {
+		updatedPhase := updateNonAdminPhase(&nab.Status.Phase, nacv1alpha1.NonAdminBackupPhaseDeleting)
+		updatedCondition := meta.SetStatusCondition(&nab.Status.Conditions,
+			metav1.Condition{
+				Type:    string(nacv1alpha1.NonAdminConditionDeleting),
+				Status:  metav1.ConditionTrue,
+				Reason:  "DeletionPending",
+				Message: "backup deletion requires VeleroBackup deletion or finalizer removal",
+			},
+		)
+		if updatedPhase || updatedCondition {
+			if err := r.Status().Update(ctx, nab); err != nil {
+				logger.Error(err, statusUpdateError)
+				return false, err
+			}
+			logger.V(1).Info("NonAdminBackup status marked for deletion")
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// createVeleroDeleteBackupRequest initiates deletion of the associated VeleroBackup object
+// that is referenced by the NameUUID within the NonAdminBackup (NAB) object.
+// This ensures the VeleroBackup is deleted before the NAB object itself is removed.
+//
+// Parameters:
+//   - ctx: Context to manage request lifetime.
+//   - logger: Logger instance for logging messages.
+//   - nab: Pointer to the NonAdminBackup object to be managed.
+//
+// This function first checks if the NAB object has the finalizer. If yes, it attempts to locate the associated
+// VeleroBackup object by the UUID stored in the NAB status. If a unique VeleroBackup object is found,
+// deletion is initiated, and the reconcile loop will be requeued. If multiple VeleroBackup objects
+// are found with the same label, the function logs an error and returns.
+//
+// Returns:
+//   - A boolean indicating whether to requeue the reconcile loop,
+//   - An error if VeleroBackup deletion or retrieval fails, for example, due to multiple VeleroBackup objects found with the same UUID label.
+func (r *NonAdminBackupReconciler) createVeleroDeleteBackupRequest(ctx context.Context, logger logr.Logger, nab *nacv1alpha1.NonAdminBackup) (bool, error) {
+	if !controllerutil.ContainsFinalizer(nab, constant.NabFinalizerName) ||
```

Comment:
I think this is not needed

this is only called after `setStatusAndConditionForDeletionAndCallDelete` function. If this function is reached, we know NAB has finalizer (and UUID)